projects/01-pair-matching/hanie-abutalebi/src/main.js

```diff
@@ -0,0 +1,115 @@
+    const original = newArray[index];
+    newArray[index] = newArray[randomIndex];
+    newArray[randomIndex] = original;
+  }
+  return newArray;
+};
+const selectRandom = (array, items) => {
+  const newArray = [...array];
+  const randomPicks = [];
+
+  for (let index = 0; index < items; index += 1) {
+    const randomIndex = Math.floor(Math.random() * newArray.length);
+    randomPicks.push(newArray[randomIndex]);
+    newArray.splice(randomIndex, 1);
+  }
+  return randomPicks;
+};
+
+const generateGame = () => {
+  // const dimensions = selectClass.board.getAttribute('data-dimension');
+  // if (dimensions % 2 !== 0) {
+  // alert('The dimension should be an even number.Please check.');
+  // }
+  const emojis = ['🌂', '🧶', '🎓', '🐹', '🐳', '🌴', '🦦', '🏹', '🎹', '🪐', '🦴', '🧠', '🧵'];
+  const picks = selectRandom(emojis, (dimNumber * dimNumber) / 2);
+  const items = shuffle([...picks, ...picks]);
+  selectClass.board_box.innerHTML = `
+    <div class="board" style="grid-template-columns: repeat(${dimNumber}, auto)">
+          ${items
+    .map(
+      (item) => `
+    <div class="card">
+      <div class="card-front"></div>
+      <div class="card-back">${item}</div>
+    </div>`,
+    ).join('')}
+     </div>
+  `;
+};
+//  flipback card
+const flipBack = () => {
+  document.querySelectorAll('.card:not(.equal)').forEach((card) => {
+    card.classList.remove('flipped');
+  });
+  flippedCards = 0;
+};
+
+//  win function
+const win = () => {
+  selectClass.win.innerHTML = `
+            <span class="win-inner">
+              congratulations ! <br /> You won! <br />
+            </span>
+        `;
+};
+
+const flipCard = (card) => {
+  flippedCards += 1;
+  // less than 2 card fliped
+  if (flippedCards <= 2) {
+    card.classList.add('flipped');
+  }
+  //  exactly 2 card fliped
+  if (flippedCards === 2) {
+    flippedCards = document.querySelectorAll('.flipped:not(.match)');
+
+    if (flippedCards[0].innerText === flippedCards[1].innerText) {
+      flippedCards[0].classList.add('equal');
+      flippedCards[1].classList.add('equal');
+    }
+    setTimeout(() => {
+      flipBack();
+    }, 1000);
+  }
+  // all cards fliped
+  if (!document.querySelectorAll('.card:not(.flipped)').length) {
+    setTimeout(() => {
+      win();
+    }, 500);
+  }
+};
+// event listener and restart button
+const setEventListeners = () => {
+  document.addEventListener('click', (event) => {
+    if (event.target.className.includes('card-front')) {
+      flipCard(event.target.parentElement);
+    } else if (event.target.className.includes('restart')) {
+      if (!document.querySelectorAll('.card:not(.flipped)').length) {
+        selectClass.win.innerHTML = '';
+      }
+      generateGame(dimNumber);
+    }
+  });
+};
+generateGame(dimNumber);
+setEventListeners(dimNumber);
+fetch('http://192.168.43.126:8000/playgame')
+  .then((response) => response.json()).then((data) => {
+    dimNumber = data;
```

Comment:
Not sure about this one, but I think everywhere in your code (all the code you have written above) you used `dimNumber` as a number. but here in this scope you are assigning an object to `dimNumber` instead of the number.
This may be a bug. Please check it.